File: routers/borrows.py
from typing import List

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from database import get_db
from schemas import Borrow, BorrowCreate
from services import borrow_service
from security import require_admin

router = APIRouter()

@router.post("/borrow",response_model=Borrow)
def borrow ( data:BorrowCreate, db: Session = Depends(get_db), user=Depends(require_admin)):
    return borrow_service.borrow(db, data.book_id, data.reader_id)

@router.post("/borrows/{borrow_id}/return",response_model=Borrow)
def return_book ( borrow_id:int, db: Session = Depends(get_db), user=Depends(require_admin)):
    return borrow_service.return_book(db, borrow_id)

@router.get("/borrows", response_model=List[Borrow])
def list_borrows(reader_id: int=None , status: str=None , db: Session = Depends(get_db), user=Depends(require_admin)):
    return borrow_service.list_borrows(db, reader_id, status)

File: frontend/src/api/borrows.js
import request from './request'

// 借阅相关的接口
export function listBorrows(params) {
  return request.get('/borrows', { params })
}

// 借书:传入图书 id 和读者 id
export function borrowBook(bookId, readerId) {
  return request.post('/borrow', { book_id: bookId, reader_id: readerId })
}

// 还书:传入借阅记录 id
export function returnBook(id) {
  return request.post(`/borrows/${id}/return`)
}
